### DISCUSSION TITLE
How to re-translate when using machine translator?

### DISCUSSION THREAD

**Discussion by leonh** [wagtail/wagtail-localize#371] [Ideas]
My intention is to use the DummyTranslator example on my wagtail site as a basis for creating a custom class of BaseMachineTranslator

I can translate the content with DummyTranslator via the admin edit page but after translation I cannot re-run the translator . How can I revert back to the initial aliased version of the page and then re-translate using an updated DummyTranslator (or any other subclass of BaseMachineTranslator) ?

**Comment by leonh** [comment]
Adding my workaround for this - in case it helps anyone else wanting to re-translate content using a machine translator.

in wagtail_localize/views/edit_translation.py

comment out this section at line 957

```
        #if StringTranslation.objects.filter(
        #    translation_of_id=string_segment.string_id,
        #    locale=translation.target_locale,
        #    context_id=string_segment.context_id,
        #).exists():
        #    continue
```

in line 972 of wagtail_localize/views/edit_translation.py

change StringTranslation.**get_or_create(....)** to StringTranslation.**update_or_create(...)**

**Reply by kaedroho** [reply]
Thanks for reporting this!

Maybe we could add a dropdown option to the "translate with machine translator" button to make it force retranslate any strings that were previously translated with a machine translator?

**Reply by leonh** [reply]
Yes that would be good....also there is the possibility of adding a ALLOW_RETRANSLATE setting when you define the machine translator class?

**Reply by kaedroho** [reply]
To allow this feature to be disabled? (I guess to stop editors from using up quotas?)

Yep, I think that makes sense

**Reply by leonh** [reply]
Yes my use case has been driven from translation of **en** to **en-US** which in our limited bounded seo content meant a bunch of replacing UK spelling with US spelling - no need for restructuring of sentences. I used the machine translator and as way to bundle-up these into one action. So it's internal service rather than an external paid for service.
